File: generate_all_682_players.py
import csv

def clean_name(name):
    """Clean player name"""
    return name.strip() if name else ""

def determine_player_type(country, cap_status):
    """Determine player type based on country and capped status"""
    country = country.strip() if country else ""
    cap_status = cap_status.strip() if cap_status else ""
    
    is_indian = country == "India"
    is_uncapped = cap_status == "Uncapped"
    
    if is_indian and is_uncapped:
        return "Indian Uncapped"
    elif is_indian:
        return "Indian"
    elif is_uncapped:
        return "Overseas Uncapped"
    else:
        return "Overseas"

def map_role(specialism):
    """Map specialism to player role"""
    spec = specialism.upper() if specialism else ""
    
    if "WICKETKEEPER" in spec or "WICKET-KEEPER" in spec:
        return "Wicket-Keeper"
    elif "BATTER" in spec or "BATSMAN" in spec:
        return "Batsman"
    elif "BOWLER" in spec:
        return "Bowler"
    elif "ALL-ROUNDER" in spec:
        return "All-Rounder"
    else:
        return "All-Rounder"

def determine_auction_group(base_price):
    """Classify players into groups A,B,C,D based on base price"""
    if base_price >= 20000000:
        return 'A'
    elif base_price >= 12500000:
        return 'B'
    elif base_price >= 7500000:
        return 'C'
    else:
        return 'D'

def escape_sql_string(s):
    """Escape single quotes in SQL strings"""
    return s.replace("'", "''") if s else ""

# Read CSV and generate SQL
sql_statements = []
player_count = 0
group_counts = {'A': 0, 'B': 0, 'C': 0, 'D': 0}

csv_path = r'c:\xampp\htdocs\ipl_auction\1731674068078_TATA IPL 2025- Auction List -15.11.24.csv'

with open(csv_path, 'r', encoding='utf-8') as f:
    reader = csv.reader(f)
    
    for row_num, row in enumerate(reader):
        # Skip first 4 header rows
        if row_num < 4:
            continue
        
        try:
            # Parse columns (using 0-based indexing)
            first_name = clean_name(row[3]) if len(row) > 3 else ""
            surname = clean_name(row[4]) if len(row) > 4 else ""
            country = row[5] if len(row) > 5 else "India"
            age_str = row[8] if len(row) > 8 else "25"
            specialism = row[9] if len(row) > 9 else ""
            cap_status = row[20] if len(row) > 20 else "Capped"
            price_rs_str = row[21] if len(row) > 21 else ""
            
            # Skip empty rows
            if not first_name or not surname:
                continue
            
            # Player Name
            player_name = escape_sql_string(f"{first_name} {surname}")
            
            # Player Type
            player_type = determine_player_type(country, cap_status)
            
            # Player Role
            player_role = map_role(specialism)
            
            # Base Price
            try:
                base_price = int(float(price_rs_str) * 100000)  # Convert lakhs to rupees
            except:
                base_price = 20000000
            
            # Auction Group
            auction_group = determine_auction_group(base_price)
            
            # Nationality
            nationality = escape_sql_string(country.strip() if country else "India")
            
            # Age
            try:
                age = int(age_str)
            except:
                age = 25
            
            # Generate SQL INSERT statement
            sql = f"INSERT INTO players (player_name, player_type, player_role, base_price, auction_group, nationality, age) VALUES ('{player_name}', '{player_type}', '{player_role}', {base_price}, '{auction_group}', '{nationality}', {age});"
            sql_statements.append(sql)
            
            # Update counts
            player_count += 1
            group_counts[auction_group] += 1
                
        except Exception as e:
            print(f"Error processing row {row_num + 1}: {e}")
            continue

# Write to SQL file
output_path = r'c:\xampp\htdocs\ipl_auction\database\all_682_players.sql'

with open(output_path, 'w', encoding='utf-8') as f:
    f.write(f"-- IPL 2025 Auction - All Players\n")
    f.write(f"-- Total Players: {player_count}\n")
    f.write(f"-- Group A (≥20 Cr): {group_counts['A']} players\n")
    f.write(f"-- Group B (12.5-20 Cr): {group_counts['B']} players\n")
    f.write(f"-- Group C (7.5-12.5 Cr): {group_counts['C']} players\n")
    f.write(f"-- Group D (<7.5 Cr): {group_counts['D']} players\n")
    f.write(f"-- Generated on: 2025-12-30\n\n")
    
    for sql in sql_statements:
        f.write(sql + "\n")

print(f"✓ Successfully processed {player_count} players")
print(f"\nDistribution across auction groups:")
print(f"  Group A (≥20 Cr):      {group_counts['A']} players")
print(f"  Group B (12.5-20 Cr):  {group_counts['B']} players")
print(f"  Group C (7.5-12.5 Cr): {group_counts['C']} players")
print(f"  Group D (<7.5 Cr):     {group_counts['D']} players")
print(f"\nSQL file saved to: {output_path}")

File: generate_all_682_players.php
<?php
// IPL 2025 Auction - Generate SQL for all 682 players

function cleanName($name) {
    return trim($name);
}

function determinePlayerType($country, $capStatus) {
    $country = trim($country);
    $capStatus = trim($capStatus);
    
    $isIndian = $country === "India";
    $isUncapped = $capStatus === "Uncapped";
    
    if ($isIndian && $isUncapped) {
        return "Indian Uncapped";
    } elseif ($isIndian) {
        return "Indian";
    } elseif ($isUncapped) {
        return "Overseas Uncapped";
    } else {
        return "Overseas";
    }
}

function mapRole($specialism) {
    $spec = strtoupper(trim($specialism));
    
    if (strpos($spec, "WICKETKEEPER") !== false || strpos($spec, "WICKET-KEEPER") !== false) {
        return "Wicket-Keeper";
    } elseif (strpos($spec, "BATTER") !== false || strpos($spec, "BATSMAN") !== false) {
        return "Batsman";
    } elseif (strpos($spec, "BOWLER") !== false) {
        return "Bowler";
    } elseif (strpos($spec, "ALL-ROUNDER") !== false) {
        return "All-Rounder";
    } else {
        return "All-Rounder";
    }
}

function determineAuctionGroup($basePrice) {
    if ($basePrice >= 20000000) { // >= 2 Crore
        return 'A';
    } elseif ($basePrice >= 10000000) { // 1 to < 2 Crore
        return 'B';
    } else { // < 1 Crore
        return 'C';
    }
}

function escapeSqlString($str) {
    return str_replace("'", "''", $str);
}

// Read CSV file
$csvPath = __DIR__ . '/1731674068078_TATA IPL 2025- Auction List -15.11.24.csv';
$outputPath = __DIR__ . '/database/all_682_players.sql';

$sqlStatements = [];
$playerCount = 0;
$groupCounts = ['A' => 0, 'B' => 0, 'C' => 0];

if (($handle = fopen($csvPath, "r")) !== FALSE) {
    $rowNum = 0;
    
    while (($row = fgetcsv($handle, 10000, ",")) !== FALSE) {
        $rowNum++;
        
        // Skip first 4 header rows
        if ($rowNum <= 4) {
            continue;
        }
        
        try {
            // Detect the CSV format by checking if column 2 is empty
            // Format 1: Has Set number in column 2, prices at columns 19/20
            // Format 2: Empty column 2, prices at columns 17/18
            $hasSetNumber = !empty(trim($row[2]));
            
            if ($hasSetNumber) {
                // Format 1: Original format
                $firstName = isset($row[3]) ? cleanName($row[3]) : "";
                $surname = isset($row[4]) ? cleanName($row[4]) : "";
                $country = isset($row[5]) ? trim($row[5]) : "India";
                $ageStr = isset($row[8]) ? trim($row[8]) : "25";
                $specialism = isset($row[9]) ? trim($row[9]) : "";
                $capStatus = isset($row[19]) ? trim($row[19]) : "Capped";
                $priceRsStr = isset($row[20]) ? trim($row[20]) : "";
            } else {
                // Format 2: Modified format (from row 575 onwards)
                $firstName = isset($row[3]) ? cleanName($row[3]) : "";
                $surname = isset($row[4]) ? cleanName($row[4]) : "";
                $country = isset($row[5]) ? trim($row[5]) : "India";
                $ageStr = isset($row[8]) ? trim($row[8]) : "25";
                $specialism = isset($row[9]) ? trim($row[9]) : "";
                $capStatus = isset($row[17]) ? trim($row[17]) : "Capped";
                $priceRsStr = isset($row[18]) ? trim($row[18]) : "";
            }
            
            // Skip empty rows
            if (empty($firstName) || empty($surname)) {
                continue;
            }
            
            // Player Name
            $playerName = escapeSqlString($firstName . " " . $surname);
            
            // Player Type
            $playerType = determinePlayerType($country, $capStatus);
            
            // Player Role
            $playerRole = mapRole($specialism);
            
            // Base Price
            if (!empty($priceRsStr) && is_numeric($priceRsStr)) {
                $basePrice = (int)(floatval($priceRsStr) * 100000);
            } else {
                $basePrice = 20000000;
            }
            
            // Auction Group
            $auctionGroup = determineAuctionGroup($basePrice);
            
            // Nationality
            $nationality = escapeSqlString(!empty($country) ? trim($country) : "India");
            
            // Age
            if (!empty($ageStr) && is_numeric($ageStr)) {
                $age = (int)$ageStr;
            } else {
                $age = 25;
            }
            
            // Generate SQL INSERT statement
            $sql = "INSERT INTO players (player_name, player_type, player_role, base_price, auction_group, nationality, age) VALUES ('$playerName', '$playerType', '$playerRole', $basePrice, '$auctionGroup', '$nationality', $age);";
            $sqlStatements[] = $sql;
            
            // Update counts
            $playerCount++;
            $groupCounts[$auctionGroup]++;
            
        } catch (Exception $e) {
            echo "Error processing row $rowNum: " . $e->getMessage() . "\n";
            continue;
        }
    }
    
    fclose($handle);
}

// Write to SQL file
$output = "-- IPL 2025 Auction - All Players\n";
$output .= "-- Total Players: $playerCount\n";
$output .= "-- Group A (>= 2 Cr): {$groupCounts['A']} players\n";
$output .= "-- Group B (1 to <2 Cr): {$groupCounts['B']} players\n";
$output .= "-- Group C (< 1 Cr): {$groupCounts['C']} players\n";
$output .= "-- Generated on: " . date('Y-m-d H:i:s') . "\n\n";

// First, delete all existing players
$output .= "DELETE FROM players;\n\n";

foreach ($sqlStatements as $sql) {
    $output .= $sql . "\n";
}

file_put_contents($outputPath, $output);

echo "✓ Successfully processed $playerCount players\n\n";
echo "Distribution across auction groups:\n";
echo "  Group A (>= 2 Cr):  {$groupCounts['A']} players\n";
echo "  Group B (1 to <2 Cr):  {$groupCounts['B']} players\n";
echo "  Group C (< 1 Cr):  {$groupCounts['C']} players\n\n";
echo "SQL file saved to: $outputPath\n";
?>
